### DISCUSSION TITLE
Why MessageParameterAttribute marked as internal?

### DISCUSSION THREAD

**Discussion by MaceWindu** [CoreWCF/CoreWCF#339] [Q&A] [answered]
I'm looking into ways to migrate our existing netfx code to .net core and one of things I discovered is that `MessageParameterAttribute` is marked as internal for some reason. Which makes me wonder how you expect users to use it?

https://github.com/CoreWCF/CoreWCF/blob/d021d89c30ee0fcc8a028936a69df8ddadad3267/src/CoreWCF.Primitives/src/CoreWCF/MessageParameterAttribute.cs

**Comment by mconnew** [comment, marked as answer]
The policy is if we don't have tests, the property is internal. This one is a little more interesting as we have some compatibility code to support using the System.ServiceModel namespaced MessageXXXXXXAttribute attributes so you can reference the WCF client nuget packages and share your contract between them. This one was missed as part of that work so just making it public is insufficient as we need to add that compat code in too. I'll create an issue to track this work.

**Reply by mconnew** [reply]
I've opened issue #345 to track this
